app/models/concerns/episode_media.rb

```diff
@@ -4,14 +4,93 @@ module EpisodeMedia
   extend ActiveSupport::Concern
 
   included do
+    enum :medium, [:audio, :uncut, :video, :override], prefix: true
+
     # NOTE: this just-in-time creates new media versions
     # TODO: convert to sql, so we don't have to load/check every episode?
     # TODO: stop loading non-latest media versions
     scope :feed_ready, -> { includes(media_versions: :media_resources).select { |e| e.feed_ready? } }
+
+    validate :validate_media_ready, if: :strict_validations
+
+    after_save :destroy_out_of_range_contents, if: ->(e) { e.segment_count_previously_changed? }
+    after_save :analyze_external_media
+  end
+
+  def validate_media_ready
+    return unless published_at.present? && media?
+
+    # media must be complete on _initial_ publish
+    # otherwise - having files in any status is good enough
+    unless enclosure_ready?(published_at_was.blank?)
+      errors.add(:base, :media_not_ready, message: "media not ready")
+    end
+  end
+
+  def enclosure_ready?(must_be_complete = true)
+    if override?
+      override_ready?(must_be_complete)
+    else
+      media_ready?(must_be_complete)
+    end
+  end
+
+  def medium=(new_medium)
+    super
+
+    if medium_changed? && medium_was.present?
+      if medium_was == "uncut" && medium == "audio"
+        uncut&.mark_for_destruction
+      elsif medium_was == "audio" && medium == "uncut"
+        if (c = contents.first)
+          build_uncut.tap do |u|
+            u.file_size = contents.first.file_size
+            u.duration = contents.first.duration
+
+            # use the feeder cdn url for older completed files
+            is_old = (Time.now - c.created_at) > 24.hours
+            u.original_url = (c.status_complete? && is_old) ? c.url : c.original_url
+          end
+        end
+        contents.each(&:mark_for_destruction)
+      else
+        contents.each(&:mark_for_destruction)
```

Comment:
What happens when you switch from (or to) medium=override?

Should this be deleting the override you changed from override->contents or something?

Reply:
I guess if it just left the override there... users would at least see it on the page.  And realize they should delete that override_url.  Not terrible.

Reply:
since an override url can exist without a type, changing the type should not delete it - I think this logical?

Reply:
+1